redis/src/types.rs

```diff
@@ -1427,6 +1494,22 @@ macro_rules! from_vec_from_redis_value {
                     _ => invalid_type_error!(v, "Response type not vector compatible."),
                 }
             }
+            fn from_owned_redis_value(v: Value) -> RedisResult<$Type> {
+                match v {
+                    // All binary data except u8 will try to parse into a single element vector.
+                    // u8 has its own implementation of from_byte_vec.
+                    Value::Data(bytes) => match FromRedisValue::from_byte_vec(&bytes) {
```

Comment:
Yeah, this is one of the limitations of the current design. The hard part here was preserving the error message- if there's a decoding error, it's supposed to show the original bytes, but if we pass them in to this/an equivalent function by value, we lose them.

It might be possible by reworking the signature for this function, but then this would become a breaking change for implementors of `FromRedisValue` outside of this crate.

Reply:
I've added a `from_owned_byte_vec` - it will slightly change the error message in the event of a failed decode, but I think there will still be enough context to figure out what's going on.